lib/puppet/type/cisco_ospf_area.rb

```diff
@@ -0,0 +1,176 @@
+      stub_no_summary => true,
+    }
+  "
+
+  ensurable
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  # Parse out the title to fill in the attributes in these
+  # patterns. These attributes can be overwritten later.
+  def self.title_patterns
+    identity = ->(x) { x }
+    patterns = []
+
+    # Below pattern matches both parts of the full composite name.
+    patterns << [
+      /^(\S+) (\S+) (\S+)$/,
+      [
+        [:ospf, identity],
+        [:vrf, identity],
+        [:area, identity],
+      ],
+    ]
+    patterns
+  end
+
+  # Overwrites name method. Original method simply returns self[:name],
+  # which is no longer valid or complete.
+  # Would not have failed, but just return nothing useful.
+  def name
+    "#{self[:ospf]} #{self[:vrf]} #{self[:area]}"
+  end
+
+  newparam(:name) do
+    desc 'Name of cisco_ospf_area, not used, but needed for puppet'
+  end
+
+  newparam(:area, namevar: true) do
+    desc 'Name of the resource instance. Valid values are string.'
+    munge do |value|
+      value = IPAddr.new(value.to_i, Socket::AF_INET) unless
+        value[/\./]
+      value
+    end
+  end # param area
+
+  newparam(:vrf, namevar: true) do
+    desc "Name of the vrf instance. Valid values are string. The
+          name 'default' is a valid VRF."
+  end # param vrf
+
+  newparam(:ospf, namevar: true) do
+    desc 'Name of the ospf instance. Valid values are string.'
+  end # param ospf
+
+  ##############
+  # Attributes #
+  ##############
+
+  newproperty(:authentication) do
+    desc 'Enable authentication for the area.'
+
+    newvalues(:clear_text, :md5, :default)
+  end # property authentication
+
+  newproperty(:default_cost) do
+    desc "default_cost for default summary Link-State Advertisement (LSA).
+          Valid values are integer, keyword 'default'."
+
+    munge { |value| value == 'default' ? :default : Integer(value) }
+  end # property default_cost
+
+  newproperty(:filter_list_in) do
+    desc "Filter networks sent to this area. Valid values are string,
+         keyword 'default'. "
+
+    munge { |value| value == 'default' ? :default : value }
+  end # property filter_list_in
+
+  newproperty(:filter_list_out) do
+    desc "Filter networks sent from this area. Valid values are string,
+         keyword 'default'. "
+
+    munge { |value| value == 'default' ? :default : value }
+  end # property filter_list_out
+
+  newproperty(:range, array_matching: :all) do
+    format = '[[ip, not_advertise, cost], [ip, na, co]]'
+    desc 'An array of [ip, not_advertise, cost] pairs. '\
+         "Valid values match format #{format}."
+
+    # Override puppet's insync method, which checks whether current value is
+    # equal to value specified in manifest.  Make sure puppet considers
+    # 2 arrays with same elements but in different order as equal.
+    def insync?(is)
+      (is.size == should.size && is.sort == should.sort)
+    end
+
+    def should_to_s(value)
```

Comment:
Please add a comment to explain why these are overridden. You told me once and it made sense but I can't remember what problem this solves. :-\